ceph/rados/core_workflows.py

```diff
@@ -1264,3 +1272,44 @@ def get_osd_stat(self):
         osd_stats = self.run_ceph_command(cmd=cmd)
         log.debug(f" The OSD Statistics are : {osd_stats}")
         return osd_stats
+
+    def get_pgid(
+        self,
+        pool_name: str = None,
+        pool_id: int = None,
+        osd: int = None,
+        osd_primary: int = None,
+    ) -> list:
+        """
+        Retrieves all the PG IDs for a pool or PG IDs where a
+        certain osd is primary in the acting set or PG IDs which are
+        utilizing the concerned osd
+        Args:
+            pool_name: name of the pool
+            pool_id: pool id
+            osd: osd id whose pgs are to be retrieved
+            osd_primary: primary osd id whose pgs are to be retrieved
```

Comment:
Please adda note stating that only one of the options need to be provided. Even if i provide all, I would get return only for ` osd_primary `.  Let's hilight that only one of them needs to be sent at a time.